### DISCUSSION TITLE
data_and_controls

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/labview_book#35] [Announcements]
# 控件的局部变量和属性

https://lv.qizhen.xyz/data_and_controls

**Comment by GRIT-LRJ** [comment]
老师您好，最后的练习怎么匹配颜色空间的颜色呢

**Reply by ruanqizhen** [reply]
这个问题是为让大家熟悉一下控件的属性，到控件的属性节点里找一下，看看能不能找到控制文字颜色相关的属性。

如果需要，可以参考下面的程序：

![](https://raw.githubusercontent.com/ruanqizhen/labview_book/main/code/_assistant/set_text_color.png)

![](https://raw.githubusercontent.com/ruanqizhen/labview_book/main/code/_assistant/set_text_color.gif)

**Reply by GRIT-LRJ** [reply]
> 
谢谢老师答疑解惑，看了老师的程序就明白了，之前一直没找到颜色盒，原来是在右键菜单里替换啊，再次感谢老师！

**Reply by whchrome** [reply]
阮老师，您好，我在学习别人的程序时发现，有一些隐藏的属性节点，比如MasterRect等，这些属性节点藏在什么地方，怎样找到？

**Reply by ruanqizhen** [reply]
> 阮老师，您好，我在学习别人的程序时发现，有一些隐藏的属性节点，比如MasterRect等，这些属性节点藏在什么地方，怎样找到？

MasterRect 是 VI Scripting 的功能之一，需要开启 VI Scripting。可以参考 [VI 脚本授权](https://lv.qizhen.xyz/vi_server_for_vi)

**Reply by whchrome** [reply]
哇，找到了找到了，非常感谢

**Comment by Anan202574** [comment]
老师，列表框例子里，项目名创建数组是二维数组不能连接到属性节点里的项目名，还有就是项目符号也不能连接到二维数组里.

**Reply by ruanqizhen** [reply]
能说的具体一点吗？最好贴个图给你的程序。

**Reply by Anan202574** [reply]
就是我在使用罗列文件夹的时候，文件夹名和文件名两个字符串生成数组以后，就是二维数组。二维数组不能作为列表框的项目名字.

**Reply by ruanqizhen** [reply]
你是说，就是使用书里的示例，但现在不工作了？不知道是不是 LabVIEW 改了什么。我现在手头没有 LabVIEW，我找时间下载一个试试。

**Reply by ruanqizhen** [reply]
> 就是我在使用罗列文件夹的时候，文件夹名和文件名两个字符串生成数组以后，就是二维数组。二维数组不能作为列表框的项目名字.

我想我明白你说的是什么意思了，你如果是想先显示文件夹再在下面显示所有文件，那么要使用 “连接输入”， 参考： https://lv.qizhen.xyz/data_array#数组合并

**Reply by Anan202574** [reply]
感谢老师，学了后面忘了前面。我把数组那里的设置给忘了。感谢指点

**Reply by Anan202574** [reply]
感谢老师，还是自己学的太粗浅，现在好了。感谢老师的指点。
